### PR TITLE
Submit cart

### PR DESCRIPTION
**Pull request by at-vyle**
## Description
- Submit cart
## Reference Requirements
- Task ID: TI-1

## Reference Isuees
#{issue_id}

## Migrations
YES | NO

## Impacted Areas in Application
List general components of the application that this PR will affect:

-

## SQL query and `explain`
List of all queries and attach the result of each explaining query (in text, screenshot...)

- Refer to [Using Explain with MySQL](https://viblo.asia/p/su-dung-explain-de-toi-uu-cau-lenh-mysql-BYjv44gmvxpV)

### CODE REVIEW

**Comment by at-hienpham** [comment]
LGTM

File: app/Http/Controllers/Api/User/OrderController.php [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    * Create order
+    *
+    * @param App\Http\Requests\CreateOrderRequest $request request
+    *
+    * @return \Illuminate\Http\Response
+    */
+    public function store(CreateOrderRequest $request)
+    {
+        $user = Auth::user();
+        $products = json_encode($request->products);//return string
```
Comment (at-quangtran) [inline app/Http/Controllers/Api/User/OrderController.php]: đoạn này encode, xong dưới lại decode ?

File: public/js/public/checkout.js [hunk 2]
```diff
@@ -13,5 +13,48 @@ $(document).ready(function() {
...
+        $.ajax({
+            url: "/api/orders",
+            type: "post",
+            headers: {
+                'Accept': 'application/json',
+                'Authorization': 'Bearer ' + token
+            },
+            data: {
+                'products' : cart,
+                'address' : $('#DIACHI').val(),
```
Comment (at-quangtran) [inline public/js/public/checkout.js]: tiếng anh

File: public/js/public/cart.js [hunk 3]
```diff
@@ -18,20 +18,21 @@ function addToCart(idProduct) {
     var name = $('#item-product-' + idProduct +' .item .item-name a h2').text();
     var price = $('#item-product-' + idProduct +' .item .item-name .price span').text();
     var cartItem;
+    console.log(img + name + price);
```
Comment (at-quangtran) [inline public/js/public/cart.js]: console

File: public/js/public/checkout.js [hunk 4]
```diff
@@ -13,5 +13,49 @@ $(document).ready(function() {
...
+            success: function(response) {
+                if(response.code == 200) {
+                    $('#modalMessageCart h3').html(Lang.get('user/cart.thank_you'));
+                    $('#modalMessageCart #modal-message').html(Lang.get('user/cart.message_susscess'));
+                    ModalMessageCart();
+                    modifyCart(0, 'clear')
+                }
+            },
+            error: function (response) {
+                console.log(response);
```
Comment (at-quangtran) [inline public/js/public/checkout.js]: console

**Pull request merged**